dockstore-webservice/src/main/java/io/dockstore/webservice/resources/AbstractHostedEntryResource.java

```diff
@@ -394,6 +394,12 @@ public T deleteHostedVersion(@ApiParam(hidden = true) @Parameter(hidden = true,
         checkEntry(entry);
         checkUserCanUpdate(user, entry);
         checkHosted(entry);
+
+        Optional<U> deleteVersion =  entry.getWorkflowVersions().stream().filter(v -> Objects.equals(v.getName(), version)).findFirst();
+        if (deleteVersion.isPresent() && deleteVersion.get().isFrozen()) {
```

Comment:
If the version isn't present, it will silently fail as well. Fail less severely than the bug you're fixing, but I think you should throw a 404 in that case.

Reply:
Oops, forgot about this. Changed